File: lab_3_assignment_3/gcd_recursive_implementation.c
#include <stdio.h>
int iterative_gcd(int, int);
int recursive_gcd(int, int);
int main()
{
    int a, b;
    printf("Enter two numbers a,b: ");
    scanf("%d%d", &a, &b);
    printf(" Iterative gcd :%d\n Recursive gcd : %d \n", iterative_gcd(a, b), recursive_gcd(a, b));

    return 0;
}
int iterative_gcd(int a, int b)
{
    int r;
    if (b == 0)
    {
        return a;
    }
    else
    {
        while (b != 0)
        {
            r = a % b;
            a = b;
            b = r;
        }
    }
    return a;
}
int recursive_gcd(int a, int b)
{
    if (b == 0)
    {
        return a;
    }
    else
    {
        return (recursive_gcd(b, a % b));
    }
}

File: lab_3_assignment_3/gcd_recursive_implementation.py
def iterative_gcd(a, b):
    if b == 0:
        return a
    else:
        while b:
            r = a % b
            a = b
            b = r
    return a


def recursive_gcd(a, b):
    if b == 0:
        return a
    else:
        return recursive_gcd(b, a % b)


num1 = int(input("Enter first numbers a : "))
num2 = int(input("Enter second numbers b : "))
print(f"Iteraive : The GCD of {num1} and {num2} is : {iterative_gcd(num1,num2)}")
print(f"Recursive : The GCD of {num1} and {num2} is : {recursive_gcd(num1,num2)}")
